std-bits/base/src/main/java/org/enso/base/numeric/Decimal_Utils.java

```diff
@@ -0,0 +1,88 @@
+package org.enso.base.numeric;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+
+/** Utils for the Enso Decmial type. */
+public class Decimal_Utils {
+  private static final BigDecimal MIN_LONG_BIGDECIMAL = BigDecimal.valueOf(Long.MIN_VALUE);
+  private static final BigDecimal MAX_LONG_BIGDECIMAL = BigDecimal.valueOf(Long.MAX_VALUE);
+
+  public static BigDecimal fromString(String s) {
+    return new BigDecimal(s);
+  }
+
+  public static Conversion_Result fromString(String s, MathContext mc) {
+    BigDecimal bd = new BigDecimal(s, mc);
+    BigDecimal withoutMC = new BigDecimal(s);
+    return new Conversion_Result(bd, bd.compareTo(withoutMC) != 0);
+  }
+
+  public static BigDecimal fromInteger(Object o) {
+    if (o instanceof Long l) {
+      // According to the BigInteger Javadocs, valueOf is preferred "because it
+      // allows for reuse of frequently used BigDecimal values".
+      return BigDecimal.valueOf(l);
+    } else if (o instanceof BigInteger bi) {
+      return new BigDecimal(bi);
+    } else {
+      throw new IllegalArgumentException("Input must be Long or BigInteger");
+    }
+  }
+
+  public static Conversion_Result fromInteger(Object o, MathContext mc) {
+    if (o instanceof Long l) {
```

Comment:
nitpick: would it make sense to turn this into a switch?
```
switch (o) {
    case Long l -> { ... }
    case BigInteger bi -> { ... }
    default, null -> 
}
```

Reply:
Done